### PR TITLE
top query k

### LINKED ISSUE #179: update find top k filtering
yay great job last week! 🦦

This week, you’ll be updating our hierarchical filtering for our new main app questions flow. This will mostly include changes to `find_top_k_filtered` in supabase, which we call in `fetchTopK` within `query.ts` to return the 4 matches that we show the adopter during the ranking process.

First, we want to make sure it’s updated for the existing questions. First, remove all references to cols with offense data. Then, add in adopter_age_pref as one of the parameters. The logic for the age filter will involve making sure the age of the adoptee is within the range given by the `age_pref` col, which should be an int[] (e.g. [30, 70] = between 30 to 70 yrs old, inclusive on both ends). 

Here is the hierarchy for filtering:

1. age (highest priority to match),
2. gender, 
3. state, 
4. veteran status (lowest priority)

You’ll have to add in the age preference filter logic, but the rest should already be there. Also add a filter to only include adoptees where `adoptee.status = WAIT_LISTED`.  

Let me know if you have any questions + make sure you understand the existing function before changing anything! Make sure you test extensively as you make changes, either through the actual app flow or with a test call ([temporary file/useEffect](https://github.com/calblueprint/adopt-an-inmate/pull/42), just delete after). As a reminder, there’s no version control for our supabase functions so be careful editing them!

*Note: it's possible `age_pref` is instead `text[]`, in which case you will have to cast or parse for the range. We should be able to fix it in [Sam's PR](https://github.com/calblueprint/adopt-an-inmate/pull/161) but not sure if it'll be changed + merged in time.

### PR DESCRIPTION
**Pull request by niranjana2222**
[//]: # "Feel free to customize this template and the emojis to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

1. `supabase/migrations/20260412120000_find_top_k_filtered.sql`

- Drops old function 
- New function (offense removed, age added): query_embedding text, k int, adopter_gender, adopter_veteran_status, adopter_state, adopter_age_pref integer[]
- Hard filter  adoptee_vector_test.status = `WAIT_LISTED'` 
- Age preference: If adopter_age_pref has at least two elements
- Other filters: gender, state, veteran 
- Relaxation order by hierarchail filters

2. `actions/queries/query.ts`

- Removed offense and added optional adopter_age_pref 

3. `actions/matching/findMatches.ts`

- Added appData.age_pref to fetchTopK 

4. `types/database.types.ts`

- Removed adopter_offense and added added adopter_age_pref

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

1. Check the SQL for filters (in relaxation order) and that argument types match the app
2. Check other changed files for updated parameter and function calls 

`tests/findTopKFiltered.integration.test.ts` 

Edge Cases
- adopter_age_pref is missing then no age filter is applied
- Adoptee dob is null then it should be excluded everywhere

CC: @InayaY3

### CODE REVIEW

File: actions/matching/findMatches.ts [hunk 1]
```diff
@@ -8,13 +8,7 @@ import { fetchTopK } from '../queries/query';
...
 assertEnvVarExists('HF_TOKEN');
 
-/**
- * Takes in an application ID and finds four matches
- * based on the application bio stored on the database.
- *
- * Returns existing matches if the application already has matches.
- * Errors if the application bio does not exist.
- */
+///finds matches
```
Comment (jinkang-0) [inline actions/matching/findMatches.ts]: why are we replacing a specific comment about what this function does to something that just reiterates the name of the function? We should keep the more descriptive comment

File: actions/queries/query.ts [hunk 2]
```diff
@@ -8,17 +8,14 @@ import {
   RankedAdopteeMatch,
 } from '@/types/schema';
 
-/* Fetch top k (by simliaity) adoptee rows with hierarchical filtering:
- * Start with all filters applied. If no results, progressively drop filters
- * starting with state, then veteran_status, and finally gender.
- */
+//fetches top k with new filtering
```
Comment (jinkang-0) [inline actions/queries/query.ts]: ^ see above

File: supabase/migrations/20260412120000_find_top_k_filtered.sql
Comment (jinkang-0) [inline supabase/migrations/20260412120000_find_top_k_filtered.sql:1]: amazing, i don't think I've ever seen anyone use database migrations yet. We should keep this up

File: tests/findTopKFiltered.integration.test.ts
Comment (jinkang-0) [inline tests/findTopKFiltered.integration.test.ts:1]: love this - we need more tests in our codebase

File: actions/matching/findMatches.ts (hunk 1 above)
Comment (niranjana2222) [reply actions/matching/findMatches.ts]: yes, that is true! mb

File: supabase/migrations/20260412120000_find_top_k_filtered.sql
Comment (niranjana2222) [reply supabase/migrations/20260412120000_find_top_k_filtered.sql:1]: yes, I was working with this in my internship rn

**Review by InayaY3 — APPROVED** [review]
tested + lgtm. awesome sauce
